api/WebApi.ts

```diff
@@ -42,7 +42,14 @@ import os = require('os');
 import url = require('url');
 import path = require('path');
 
-const isBrowser: boolean = typeof window !== 'undefined';
+// https://www.electronjs.org/docs/api/process
+interface ElectronProcess {
+  type: 'browser' | 'renderer' | 'worker'
+};
+const isElectron: boolean = process && process.versions && process.versions.hasOwnProperty('electron');
+const isElectronWebpage: boolean = isElectron && (<ElectronProcess><unknown>process).type === 'renderer';
+const isBrowser: boolean = typeof window !== 'undefined' && (!isElectron || isElectronWebpage);
```

Comment:
Could it make sense to add null check for windows.navigator property instead of introducing electron specific logic?

Reply:
@anatolybolshakov it is not enough. Please refer to this discussion: https://github.com/electron/electron/issues/2288. Although they are recommending to check userAgent....

Reply:
I believe that the correct way for doing this is to just keep the first check 
`const isElectron: boolean = process && process.versions && process.versions.hasOwnProperty('electron');`
but remove the latter one 
`const isElectronWebpage: boolean = isElectron && (<ElectronProcess><unknown>process).type === 'renderer';`

because if it is the renderer `navigator.userAgent` will be defined.